### PR TITLE
python3Packages.gdtoolkit: Use lark_0_8_0 (split from python3Packages.lark) and enable tests

### PR DESCRIPTION
**Pull request by hesiod**
###### Description of changes

Based on my findings in https://github.com/NixOS/nixpkgs/pull/216280#issuecomment-1429521866 (gdtoolkit is broken with lark 1.1.5):
* ~~Split `python3Packages.lark` into `python3Packages.lark_0_8_0` and `python3Packages.lark_1_1_5` (`python3Packages.lark` points to `python3Packages.lark_1_1_5`)~~
* ~~Use `python3Packages.lark_0_8_0` for `python3Packages.gdtoolkit`~~

Edit: I've edited the PR based on https://github.com/NixOS/nixpkgs/pull/216329#issuecomment-1430165592:
* Move `gdtoolkit` from `pkgs/development/python-modules` to `pkgs/development/tools/gdtoolkit` (also, remove gdtoolkit from `pkgs/top-level/python-packages.nix` and add it to `pkgs/top-level/all-packages.nix`)
* Run gdtoolkit tests

PR is based on #216280.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by hesiod** [comment]
Marked as draft until #216280 is merged, but this PR is ready for review.

**Review by FRidh — CHANGES_REQUESTED** [review]
Multiple versions of the same runtime package are not allowed in the Python packages set. I see it was already the case before, that was not OK either. We need to choose *one* version.

**Comment by hesiod** [comment]
@FRidh OK, so should we downgrade `lark` to `0.8.0` (or perhaps `0.8.9` if that works with gdtoolkit)?
As mentioned, gdtoolkit is definitely broken with newer lark versions.
Are there any alternatives?

**Comment by FRidh** [comment]
Looks like `gdtoolkit` is just a set of executables. It can be moved  out of `python-packages.nix`, use `buildPythonApplication`, and then its own version of `lark`.

**Comment by hesiod** [comment]
@FRidh Good point. Not sure why this was missed in the original PR (#160040). I'll look into this.

**Comment by hesiod** [comment]
@FRidh I've implemented your suggestion. See https://github.com/NixOS/nixpkgs/pull/216329/commits/5c8a28b9276dff5f6d1ba9230108c55851e61c21
Please let me know if that works for you.

**Review by FRidh — APPROVED** [review]

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -7581,6 +7581,8 @@ with pkgs;
 
   gdmap = callPackage ../tools/system/gdmap { };
 
+  gdtoolkit = callPackage ../development/tools/gdtoolkit { };
```
Comment (kirillrdy) [inline pkgs/top-level/all-packages.nix:7584]: suggestion: add gdtoolkit to `python-aliases.nix` so that people know that it got moved

Comment (hesiod) [reply pkgs/top-level/all-packages.nix:7584]: Sure, will do!

Comment (hesiod) [reply pkgs/top-level/all-packages.nix:7584]: fixed in rebased commits

**Review by kirillrdy — APPROVED** [review]

**Review by shiryel — APPROVED** [review]
`nix-review pr 216329` building and running as expected.

**Pull request merged**